aten/src/ATen/native/cuda/AmpKernels.cu

```diff
@@ -0,0 +1,110 @@
+#define _USE_MATH_DEFINES
+
+#include <math.h>
+
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cuda/Loops.cuh>
+
+
+namespace at {
+namespace native {
+// These functions have no CPU variants, so there's no need to bother with a .cpp file in native/,
+// and also no need to impose the DECLARE/DEFINE/REGISTER_DISPATCH layer of indirection.
+
+// _amp_unscale_inf_check_cuda acts in-place on scaled_grad, multiplying each element by rscale's value.
+// If any thread finds an inf/nan, it posts a 1. to found_inf.
```

Comment:
Multiplies scaled_grad in-place by rscale. 
If an element of scaled_grad was inf or NaN sets found_inf to 1.